### PR TITLE
require timeout from external file soem_config

### PR DESCRIPTION
**Pull request by polfeliu**
Externalized the timeouts for https://github.com/bnjmnp/pysoem/pull/162

### CODE REVIEW

**Comment by bnjmnp** [comment]
Hi @polfeliu, thank you for the PR. I get the connection to your PR in pysoem.
Did you consider adding this PR to the original SOEM repository? I prefer the pysoem fork of SOEM not to drift too much apart from the original SOEM.

**Comment by polfeliu** [comment]
Hi @bnjmnp, 

I would like this but I saw that the fork already has some modifications. Don't know what they exactly do they do. I share the same though about not drifting from SOEM.
I can certainly ask if this can be merged to SOEM, but I'm pretty sure it won't be accepted there because it's a breaking change.

SOEM does not require any configuration file to the user of the libs, so anything that has to be introduced by the pre-processor externally means breaking things for other users. I've searched but I haven't found any pattern that "includes if exists" for C. 

So, from my point of view, this cannot be implemented without a breaking change.
FYI regarding this there's an issue open since 2016: https://github.com/OpenEtherCATsociety/SOEM/issues/65

**Comment by bnjmnp** [comment]
Okay, you are right. And it is not a very big change, so the patch can be applied to newer versions of SOEM easily.

File: soem/ethercattype.h [hunk 1]
```diff
@@ -24,6 +24,7 @@ extern "C"
 #endif
 
 #include "osal.h"
+#include "soem_config.h"
```
Comment (bnjmnp) [inline soem/ethercattype.h:27]: Maybe we could add a compile switch here:
```C
#ifdef USE_SOEM_CONFIG
#include "soem_config.h"
#else
/** timeout value in us for tx frame to return to rx */
#define EC_TIMEOUTRET      2000
/** timeout value in us for safe data transfer, max. triple retry */
#define EC_TIMEOUTRET3     (EC_TIMEOUTRET * 3)
/** timeout value in us for return "safe" variant (f.e. wireless) */
#define EC_TIMEOUTSAFE     20000
...
#endif
```
With this compiling SOEM would not fail by default.

**Pull request merged**